### PR TITLE
Product page links

### PR DESCRIPTION
**Pull request by kuiche**
This adds a tab to the product edit screen displaying product pages for easier navigation.

Requires https://github.com/mothership-ec/cog-mothership-commerce/pull/453.

### CODE REVIEW

**Comment by thomasjthomasj** [comment]
A logic exception gets thrown (`Service `product.page_type.listing` must be defined within the installation`) when trying to add a new page on existing installations. I think this exception must have been included with the product upload stuff but that side of stuff is inaccessible. I think ideally there will be some kind of default setup for this service, and less ideally, you won't have access to page creation from this tab without this service being set up. If we go for that option then obviously we can add a service to the base installation but it would be nice if it could work straight off the bat as soon as someone upgrades their MS installation

**Comment by kuiche** [comment]
It's not using that listing class anywhere (it creates them under whatever parent you give) so it should be possible to remove that requirement when just creating single pages.

**Comment by thomasjthomasj** [comment]
Okay well we need to make it possible to do that without that exception being thrown then i guess

File: src/Controller/ProductPage/Create.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+			}
+
+			$page = $pageCreate->create($product, [
+				Options::PARENT => $data['parent'],
+				Options::LISTING_TYPE => OPTIONS::SHOP,
+				Options::PAGE_VARIANTS => $data['option_name'],
+			], $unit, $data['option_name']);
+
+			if(!$page) {
+				$this->addFlash('error', 'Product page could not be created. Does a page for this product already exist?');
```
Comment (thomasjthomasj) [inline src/Controller/ProductPage/Create.php]: Use translation key

File: src/ProductPage/Create.php [hunk 2]
```diff
@@ -85,13 +110,17 @@ public function __construct(
...
 		PageType\Collection $pageTypes,
-		PageType\PageTypeInterface $listingPageType,
+		$listingPageType,
 		ProductPageCreateEventDispatcher $dispatcher,
 		ParentPageCreateEventDispatcher $parentDispatcher,
 		Exists $exists,
 		array $productPageTypeMapping
 	)
 	{
+		if ($listingPageType !== null && $listingPageType instanceof PageType\PageTypeInterface) {
```
Comment (thomasjthomasj) [inline src/ProductPage/Create.php]: This statement looks wrong, surely it should be `listingPageType !== null && !$listingPageType instanceof PageType\PageTypeInterface`. Need to test on product upload

**Pull request merged**